drivers/wireless/ieee80211/bcm43xxx/bcmf_core.c

```diff
@@ -328,24 +331,114 @@ int bcmf_upload_file(FAR struct bcmf_sdio_dev_s *sbus, uint32_t address,
 
 int bcmf_upload_nvram(FAR struct bcmf_sdio_dev_s *sbus)
 {
-  int ret;
+  FAR uint8_t *nvram_buf;
   uint32_t nvram_sz;
   uint32_t token;
+  int ret;
+
+#ifdef CONFIG_IEEE80211_BROADCOM_FWFILES
+  const char *nvfile = CONFIG_IEEE80211_BROADCOM_NVFILENAME;
+  unsigned char swap[32];
+  struct file finfo;
+  struct stat sbuf;
+  FAR uint8_t *buf;
+  bool bskipline;
+  int i;
+
+  nvram_buf = NULL;
+  nvram_sz = 0;
+
+  if (strlen(nvfile) <= 0)
+    {
+      goto out_ramload;
+    }
+
+  ret = file_open(&finfo, nvfile, O_RDONLY);
+  if (ret < 0)
+    {
+      goto out_ramload;
+    }
+
+  ret = file_fstat(&finfo, &sbuf);
+  if (ret < 0 || sbuf.st_size <= 0)
+    {
+      goto out_ramload_with_file;
+    }
+
+  /* Round up the ram buffer size */
+
+  sbuf.st_size = (sbuf.st_size + 63) & (-64);
```

Comment:
why change  sbuf.st_size

Reply:
allocated buffer need align to the 64 for upload binary request